app/controllers/sessions_controller.rb

```diff
@@ -26,4 +26,22 @@ def destroy
     log_out if logged_in?
     redirect_to root_url
   end
+
+  def omniauth
+    user = User.find_or_create_by(uid: request.env['omniauth.auth'][:uid],
```

Comment:
`request.env['omniauth.auth']` sometimes has value `nil`. 
You should use `&.dig()`.

Reply:
em đã sửa